internal/provider/resource_consul_cluster.go

```diff
@@ -261,6 +272,29 @@ func resourceConsulClusterCreate(ctx context.Context, d *schema.ResourceData, me
 		return diag.Errorf("specified Consul version (%s) is unavailable; must be one of: [%s]", consulVersion, consul.VersionsToString(availableConsulVersions))
 	}
 
+	// If specified, validate and parse the primary link provided for federation.
+	primary_link, ok := d.GetOk("primary_link")
+	var primary *sharedmodels.HashicorpCloudLocationLink
+	if ok {
+		primary, err = parseLinkURL(primary_link.(string), ConsulClusterResourceType)
+		if err != nil {
+			return diag.Errorf(err.Error())
+		}
+		if primary.Location.OrganizationID == "" {
```

Comment:
Nit: With the implementation you have here, you can probably take this block of code out of the `primary.Location.OrganizationID == ""` conditional. `parseLinkURL()` parses the URL without attempting to resolve any additional information, so `primary.Location.OrganizationID` would always be empty.

Reply:
Done